### PR TITLE
Proper syndrome registers debugging

### PR DESCRIPTION
**Pull request by AlessandroA**
This PR makes debugging more user-friendly by explaining in a verbose way the reason for every fault.

@meriac @Patater @niklas-arm

### CODE REVIEW

File: core/debug/src/debug.c [hunk 1]
```diff
@@ -93,34 +93,126 @@ UVISOR_WEAK void debug_fault_bus_hw(void) {}
...
 {
-    dprintf("* HFSR  : 0x%08X\r\n\r\n", SCB->HFSR);
-    dprintf("* CFSR  : 0x%08X\r\n\r\n", SCB->CFSR);
-    dprintf("* DFSR  : 0x%08X\r\n\r\n", SCB->DFSR);
-    dprintf("* BFAR  : 0x%08X\r\n\r\n", SCB->BFAR);
-    dprintf("* MMFAR : 0x%08X\r\n\r\n", SCB->MMFAR);
+    dprintf("* FAULT SYNDROME REGISTERS\r\n");
+    dprintf("\r\n");
+
+    uint32_t hfsr = SCB->CFSR;
```
Comment (Patater) [inline core/debug/src/debug.c]: Should this be HFSR and not CFSR?

Comment (AlessandroA) [inline core/debug/src/debug.c]: Good catch! Fixed.

**Comment by AlessandroA** [comment]
@Patater Added whole register as well. Now you always get the full register view (HFSR, CFSR or DFSR) and then also the detailed bit-by-bit explanation. On a given fault, only bits specific to that fault are spelled out (UFSR, BFSR, MMFSR views in the CFSR).

**Comment by Patater** [comment]
Excellent. Just what the doctor ordered. Thanks!

**Pull request merged**